tests/core/full_node/test_tx_processing_queue.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from random import sample
+from secrets import token_bytes
+from typing import Dict, List
+
+import blspy
+import pytest
+
+from chia.full_node.tx_processing_queue import TransactionQueue, TransactionQueueFull
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.coin_spend import CoinSpend
+from chia.types.condition_opcodes import ConditionOpcode
+from chia.types.spend_bundle import SpendBundle
+from chia.types.transaction_queue_entry import TransactionQueueEntry
+from chia.util.ints import uint64
+from chia.wallet.puzzles.p2_delegated_puzzle_or_hidden_puzzle import solution_for_conditions
+from tests.core.make_block_generator import puzzle_hash_for_index
+
+log = logging.getLogger(__name__)
+
+
+# this code generates random spend bundles.
+
+
+def make_standard_spend_bundle(count: int) -> SpendBundle:
+    puzzle_dict: Dict[bytes32, Program] = {}
+    starting_ph: bytes32 = puzzle_hash_for_index(0, puzzle_dict)
+    solution: Program = solution_for_conditions(
+        Program.to([[ConditionOpcode.CREATE_COIN, puzzle_hash_for_index(1, puzzle_dict), uint64(100000)]])
+    )
+    coins = [Coin(bytes32(index.to_bytes(32, "big")), starting_ph, uint64(100000)) for index in range(count)]
+    coin_spends = [CoinSpend(coin, puzzle_dict[starting_ph], solution) for coin in coins]
+    spend_bundle = SpendBundle(coin_spends, blspy.G2Element())
+    return spend_bundle
+
+
+# we only want to call this once as it can take minutes if we call it many times.
+standard_spend_bundle = make_standard_spend_bundle(3)
+
+
+def get_peer_id() -> bytes32:
+    return bytes32(token_bytes(32))
+
+
+def get_transaction_queue_entry(peer_id: bytes32) -> TransactionQueueEntry:
+    sb: SpendBundle = standard_spend_bundle
+    return TransactionQueueEntry(
+        sb,
+        peer_id,  # we cheat a bit here by reusing transaction_bytes to store our peer_id
+        sb.name(),
+        None,
+        False,
+    )
+
+
+class TestTransactionQueue:
+    @pytest.fixture(scope="function")
+    def transaction_queue(self) -> TransactionQueue:
+        return TransactionQueue(1000, log)
+
+    @pytest.mark.asyncio
+    async def test_local_txs(self, transaction_queue: TransactionQueue) -> None:
```

Comment:
I don't see a test that ensures local TXs take precedence over TXs we receive from peers

Reply:
i was about to add to add that lol, good catch.